src/Sylius/Bundle/CoreBundle/Resources/config/services/promotion.xml

```diff
@@ -13,11 +13,11 @@
 
 <container xmlns="http://symfony.com/schema/dic/services" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://symfony.com/schema/dic/services http://symfony.com/schema/dic/services/services-1.0.xsd">
     <services>
-        <service id="sylius.factory.promotion_action.decorated" class="Sylius\Component\Core\Factory\PromotionActionFactory" decorates="sylius.factory.promotion_action" public="false">
-            <argument type="service" id="sylius.factory.promotion_action.decorated.inner" />
+        <service id="sylius.custom_factory.promotion_action" class="Sylius\Component\Core\Factory\PromotionActionFactory" decorates="sylius.factory.promotion_action" public="false">
+            <argument type="service" id="sylius.custom_factory.promotion_action.inner" />
         </service>
-        <service id="sylius.factory.promotion_rule.decorated" class="Sylius\Component\Core\Factory\PromotionRuleFactory" decorates="sylius.factory.promotion_rule" public="false">
-            <argument type="service" id="sylius.factory.promotion_rule.decorated.inner" />
+        <service id="sylius.custom_factory.promotion_rule" class="Sylius\Component\Core\Factory\PromotionRuleFactory" decorates="sylius.factory.promotion_rule" public="false">
```

Comment:
Why? For me `sylius.decorating_factory.promotion_rule` seems to be sophisticated enough.